src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -162,12 +162,13 @@ private async Task ProcessQueueAsync()
         {
             // Keep track of solution state modifications made by LSP requests
             Solution? lastMutatedSolution = null;
+            QueueItem work = default;
```

Comment:
What worries me this being outside the loop now is if a sequence happens like:

1. There's two items in the queue.
2. We are processing the first one. work.CallbackAsync() is called for the first item, and between that call and fetching the second item...
3. ...we throw.

If we call EnsureQueueItemCancelled twice, is that bad?

(this PR is really making me think we're better off writing this as a queue of Tasks since I think these cases then get handled for us....)